src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,50 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        if (actor == null) {
+            throw new RuntimeException("unacceptable data");
+        }
+
+        if (actor.getId() != null && get(actor.getId()).isPresent()) {
+            throw new RuntimeException(actor + "already existed in database");
+        }
+
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingExeption("adding " + actor + " into database failed");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Session session = null;
+        Actor actor = null;
+        try {
+            session = factory.openSession();
+            actor = session.get(Actor.class, id);
+            session.close();
+        } catch (Exception e) {
+            throw new DataProcessingExeption("getting actor with " + id + " from database failed");
```

Comment:
The exception thrown here uses `DataProcessingExeption`, which contains a typo. It should be `DataProcessingException` to ensure the correct exception class is used.